### PR TITLE
Add hadoop-aws jar to druid package

### PR DESCRIPTION
**Pull request by nishantmonu51**
Fixes - https://github.com/apache/incubator-druid/issues/3435

### CODE REVIEW

**Comment by ccaominh** [comment]
Would be good to add S3 docs to the HDFS page (similar to the GCS ones): https://druid.apache.org/docs/latest/development/extensions-core/hdfs.html

File: extensions-core/hdfs-storage/pom.xml [hunk 1]
```diff
@@ -255,7 +255,6 @@
           <groupId>org.apache.hadoop</groupId>
           <artifactId>hadoop-aws</artifactId>
           <version>${hadoop.compile.version}</version>
-          <scope>provided</scope>
```
Comment (ccaominh) [inline extensions-core/hdfs-storage/pom.xml]: The dependency analysis job is flagging this. I suggest changing the scope to `runtime`.

**Comment by jihoonson** [comment]
Hi @nishantmonu51, would you please fix the CI? I think [this comment](https://github.com/apache/druid/pull/8782#discussion_r351422301) will fix it.

**Comment by ccaominh** [comment]
@nishantmonu51 Are you still interested in making this fix?

**Comment by ccaominh** [comment]
https://github.com/apache/druid/pull/9399 also adds hadoop-aws

**Comment by nishantmonu51** [comment]
thanks @ccaominh for looking into it.

**Pull request closed without merging**